README.md
[![Pub](https://img.shields.io/pub/v/flutter_country_code_picker.svg)](https://pub.dartlang.org/packages/flutter_country_code_picker)

# flutter_country_code_picker

A Flutter package for showing a country code selector.

<img src="https://raw.githubusercontent.com/Salvatore-Giordano/CountryCodePicker/master/screenshots/screen1.png" width="240"/>
<img src="https://raw.githubusercontent.com/Salvatore-Giordano/CountryCodePicker/master/screenshots/screen2.png" width="240"/>

## Usage

Just put the component in your application setting the onChanged callback.

```dart

@override
 Widget build(BuildContext context) => new Scaffold(
     body: new Center(
       child: new CountryCodePicker(
         onChanged: print,
         // Initial selection and favorite can be one of code ('IT') OR dial_code('+39')
         initialSelection: 'IT',
         favorite: ['+39','FR'],
         // optional. Shows only country name and flag
         showCountryOnly: false,
         // optional. Shows only country name and flag when popup is closed.
         showOnlyCountryWhenClosed: false,
         // optional. aligns the flag and the Text left
         alignLeft: false,
       ),
     ),
 );

```

Note: Your onChanged function can be any function of the type shown below:

```dart
(CountryCode)->dynamic

```

Example:

```dart

void _onCountryChange(CountryCode countryCode) {
    //Todo : manipulate the selected country code here
    print("New Country selected: " + countryCode.toString());
  }

```

## Contributions

Contributions of any kind are more than welcome! Feel free to fork and improve country_code_picker in any way you want, make a pull request, or open an issue.


CHANGELOG.md
## 2.0.0 - unreleased

- Compressed flags to reduce package size
- Use `ListView.builder` for better performance
- Clean up code

# Original library changelog

## 1.2.4

Fixed a bug that was making impossible to update initial selection

## 1.2.3

Update country code of Republica Dominicana.

## 1.2.2

Fix code list.

## 1.2.0

Added Ability to render custom Widget instead of package one's.

Bug fix.

## 1.1.7

Flag is now optional. Fix bug on initState.

## 1.1.5

OnlyCountrymode now also displays only the country on Textwidget when closed.

## 1.1.4

Add possibility to show only country name

## 1.1.1

Update allowed dart version and modify description

## 1.1.0

Changed CElement with CountryCode and fix error on favorite null

## 1.0.4

Update country name with translated version

## 1.0.3

Update flags dimension to reduce application size

## 1.0.2

Update framework compatiblity

## 1.0.1

Correct README and update screenshots

## 1.0.0

Use png flags instead of a font

## 0.2.2

Added textStyle and padding as widget parameters

## 0.2.1

Added some documentation

## 0.2.0

Now onChanged has a full CElement as argument and not only a string. (issue #4)

## 0.1.3

Favorite and initial selection can be one of code or dial code

## 0.1.2

Favorite and initial selection can be one of code or dial code

## 0.1.1

Tested with dart 2

## 0.1.0

Removed flags in iOS because they show up weirdly.

## 0.0.2

Add favorite countries option.

## 0.0.1

Initial release


lib/src/dialog.dart
import 'package:flutter/material.dart';
import 'package:flutter_country_code_picker/src/country.dart';

/// selection dialog used for selection of the country code
class CountryPickerDialog extends StatefulWidget {
  final List<Country> countries;
  final bool showCountryOnly;
  final InputDecoration searchDecoration;
  final TextStyle searchStyle;
  final WidgetBuilder emptySearchBuilder;
  final bool showFlag;
  final List<Country> favouriteCountries;
  final String searchLabel;

  CountryPickerDialog(
    this.countries, {
    this.favouriteCountries,
    Key key,
    this.showCountryOnly,
    this.emptySearchBuilder,
    this.searchDecoration,
    this.searchStyle,
    this.showFlag,
    this.searchLabel = "Search countries",
  }) : super(key: key);

  @override
  State<StatefulWidget> createState() => _CountryPickerDialogState();
}

class _CountryPickerDialogState extends State<CountryPickerDialog> {
  /// this is useful for filtering purpose
  List<Country> filteredCountries;

  @override
  void initState() {
    filteredCountries = widget.countries;
    super.initState();
  }

  int get _listLength {
    var length = 0;
    if (widget.favouriteCountries.isNotEmpty) {
      length++;
    }

    if (filteredCountries.isEmpty) {
      length++;
    } else {
      length += filteredCountries.length;
    }

    return length;
  }

  Widget _buildListElement(BuildContext context, int index) {
    if (widget.favouriteCountries.isNotEmpty) {
      if (index == 0) {
        return Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: <Widget>[]
              ..addAll(widget.favouriteCountries
                  .map(
                    (f) => SimpleDialogOption(
                      child: _buildCountry(f),
                      onPressed: () {
                        _selectCountry(f);
                      },
                    ),
                  )
                  .toList())
              ..add(const Padding(
                padding: const EdgeInsets.symmetric(horizontal: 24),
                child: const Divider(),
              )));
      } else {
        index--;
      }
    }

    if (filteredCountries.isEmpty) {
      if (index == 0) {
        return _buildEmptySearchWidget(context);
      } else {
        index--;
      }
    }

    final country = filteredCountries[index];

    return SimpleDialogOption(
      key: Key(country.toLongString()),
      child: _buildCountry(country),
      onPressed: () {
        _selectCountry(country);
      },
    );
  }

  @override
  Widget build(BuildContext context) {
    return SimpleDialog(
      titlePadding: const EdgeInsets.fromLTRB(24, 8, 24, 0),
      title: TextField(
        style: widget.searchStyle,
        decoration: (widget.searchDecoration ?? InputDecoration()).copyWith(
          labelText: widget.searchLabel,
        ),
        onChanged: _filterElements,
      ),
      children: [
        Container(
          width: MediaQuery.of(context).size.width,
          height: MediaQuery.of(context).size.height,
          child: ListView.builder(
            itemCount: _listLength,
            itemBuilder: _buildListElement,
          ),
        ),
      ],
    );
  }

  Widget _buildCountry(Country country) {
    return Container(
      height: 32,
      child: Row(
        children: <Widget>[
          widget.showFlag
              ? Padding(
                  padding: const EdgeInsets.only(right: 16.0),
                  child: Image.asset(
                    country.flagUri,
                    package: 'flutter_country_code_picker',
                    width: 32.0,
                  ),
                )
              : Container(),
          Expanded(
            child: Text(
              country.name,
            ),
          ),
          SizedBox(width: 8),
          Text(country.dialCode)
        ],
      ),
    );
  }

  Widget _buildEmptySearchWidget(BuildContext context) {
    if (widget.emptySearchBuilder != null) {
      return widget.emptySearchBuilder(context);
    }

    return Center(child: Text('No Country Found'));
  }

  void _filterElements(String value) {
    if (value.isEmpty) {
      setState(() {
        filteredCountries = widget.countries;
      });
      return;
    }

    value = value.toUpperCase();
    setState(() {
      filteredCountries = widget.countries
          .where(
            (country) =>
                country.code.contains(value) ||
                country.dialCode.contains(value) ||
                country.name.toUpperCase().contains(value),
          )
          .toList();
    });
  }

  void _selectCountry(Country country) {
    Navigator.pop(context, country);
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:flutter_country_code_picker/flutter_country_code_picker.dart';

void main() => runApp(ExampleApp());

class ExampleApp extends StatefulWidget {
  @override
  _ExampleAppState createState() => _ExampleAppState();
}

class _ExampleAppState extends State<ExampleApp> {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
        home: Scaffold(
            appBar: AppBar(
              title: const Text('Country Picker Example'),
            ),
            body: Center(
              child: Column(
                mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                children: <Widget>[
                  CountryCodePicker(
                    onChanged: print,
                    // Initial selection and favorite can be one of code ('IT') OR dial_code ('+39')
                    initialCountry: 'IT',
                    favorites: ['+39', 'FR'],
                    //Get the country information relevant to the initial selection
                    onInit: (code) => print("${code.name} ${code.dialCode}"),
                  ),
                  SizedBox(
                    width: 400,
                    height: 60,
                    child: Padding(
                      padding: const EdgeInsets.all(8.0),
                      child: CountryCodePicker(
                        onChanged: print,
                        initialCountry: 'TF',
                        showCountryOnly: true,
                        showCountryName: true,
                        alignLeft: true,
                        builder: (countryCode) {
                          return Text('${countryCode.code}');
                        },
                      ),
                    ),
                  ),
                  SizedBox(
                    width: 400,
                    height: 60,
                    child: Padding(
                      padding: const EdgeInsets.all(8.0),
                      child: CountryCodePicker(
                          onChanged: print,
                          initialCountry: 'TF',
                          showCountryOnly: true,
                          showCountryName: true,
                          favorites: ['+39', 'FR']),
                    ),
                  ),
                ],
              ),
            )));
  }
}


lib/src/countries.dart
import 'country.dart';

const countries = <Country>[
  Country(
    name: "افغانستان",
    code: "AF",
    dialCode: "+93",
  ),
  Country(
    name: "Åland",
    code: "AX",
    dialCode: "+358",
  ),
  Country(
    name: "Shqipëria",
    code: "AL",
    dialCode: "+355",
  ),
  Country(
    name: "الجزائر",
    code: "DZ",
    dialCode: "+213",
  ),
  Country(
    name: "American Samoa",
    code: "AS",
    dialCode: "+1684",
  ),
  Country(
    name: "Andorra",
    code: "AD",
    dialCode: "+376",
  ),
  Country(
    name: "Angola",
    code: "AO",
    dialCode: "+244",
  ),
  Country(
    name: "Anguilla",
    code: "AI",
    dialCode: "+1264",
  ),
  Country(
    name: "Antarctica",
    code: "AQ",
    dialCode: "+672",
  ),
  Country(
    name: "Antigua and Barbuda",
    code: "AG",
    dialCode: "+1268",
  ),
  Country(
    name: "Argentina",
    code: "AR",
    dialCode: "+54",
  ),
  Country(
    name: "Հայաստան",
    code: "AM",
    dialCode: "+374",
  ),
  Country(
    name: "Aruba",
    code: "AW",
    dialCode: "+297",
  ),
  Country(
    name: "Australia",
    code: "AU",
    dialCode: "+61",
  ),
  Country(
    name: "Österreich",
    code: "AT",
    dialCode: "+43",
  ),
  Country(
    name: "Azərbaycan",
    code: "AZ",
    dialCode: "+994",
  ),
  Country(
    name: "Bahamas",
    code: "BS",
    dialCode: "+1242",
  ),
  Country(
    name: "‏البحرين",
    code: "BH",
    dialCode: "+973",
  ),
  Country(
    name: "Bangladesh",
    code: "BD",
    dialCode: "+880",
  ),
  Country(
    name: "Barbados",
    code: "BB",
    dialCode: "+1246",
  ),
  Country(
    name: "Белару́сь",
    code: "BY",
    dialCode: "+375",
  ),
  Country(
    name: "België",
    code: "BE",
    dialCode: "+32",
  ),
  Country(
    name: "Belize",
    code: "BZ",
    dialCode: "+501",
  ),
  Country(
    name: "Bénin",
    code: "BJ",
    dialCode: "+229",
  ),
  Country(
    name: "Bermuda",
    code: "BM",
    dialCode: "+1441",
  ),
  Country(
    name: "ʼbrug-yul",
    code: "BT",
    dialCode: "+975",
  ),
  Country(
    name: "Bolivia",
    code: "BO",
    dialCode: "+591",
  ),
  Country(
    name: "Bosna i Hercegovina",
    code: "BA",
    dialCode: "+387",
  ),
  Country(
    name: "Botswana",
    code: "BW",
    dialCode: "+267",
  ),
  Country(
    name: "Bouvetøya",
    code: "BV",
    dialCode: "+47",
  ),
  Country(
    name: "Brasil",
    code: "BR",
    dialCode: "+55",
  ),
  Country(
    name: "British Indian Ocean Territory",
    code: "IO",
    dialCode: "+246",
  ),
  Country(
    name: "Negara Brunei Darussalam",
    code: "BN",
    dialCode: "+673",
  ),
  Country(
    name: "България",
    code: "BG",
    dialCode: "+359",
  ),
  Country(
    name: "Burkina Faso",
    code: "BF",
    dialCode: "+226",
  ),
  Country(
    name: "Burundi",
    code: "BI",
    dialCode: "+257",
  ),
  Country(
    name: "Kâmpŭchéa",
    code: "KH",
    dialCode: "+855",
  ),
  Country(
    name: "Cameroon",
    code: "CM",
    dialCode: "+237",
  ),
  Country(
    name: "Canada",
    code: "CA",
    dialCode: "+1",
  ),
  Country(
    name: "Cabo Verde",
    code: "CV",
    dialCode: "+238",
  ),
  Country(
    name: "Cayman Islands",
    code: "KY",
    dialCode: "+ 345",
  ),
  Country(
    name: "Ködörösêse tî Bêafrîka",
    code: "CF",
    dialCode: "+236",
  ),
  Country(
    name: "Tchad",
    code: "TD",
    dialCode: "+235",
  ),
  Country(
    name: "Chile",
    code: "CL",
    dialCode: "+56",
  ),
  Country(
    name: "中国",
    code: "CN",
    dialCode: "+86",
  ),
  Country(
    name: "Christmas Island",
    code: "CX",
    dialCode: "+61",
  ),
  Country(
    name: "Cocos (Keeling) Islands",
    code: "CC",
    dialCode: "+61",
  ),
  Country(
    name: "Colombia",
    code: "CO",
    dialCode: "+57",
  ),
  Country(
    name: "Komori",
    code: "KM",
    dialCode: "+269",
  ),
  Country(
    name: "République du Congo",
    code: "CG",
    dialCode: "+242",
  ),
  Country(
    name: "République démocratique du Congo",
    code: "CD",
    dialCode: "+243",
  ),
  Country(
    name: "Cook Islands",
    code: "CK",
    dialCode: "+682",
  ),
  Country(
    name: "Costa Rica",
    code: "CR",
    dialCode: "+506",
  ),
  Country(
    name: "Côte d'Ivoire",
    code: "CI",
    dialCode: "+225",
  ),
  Country(
    name: "Hrvatska",
    code: "HR",
    dialCode: "+385",
  ),
  Country(
    name: "Cuba",
    code: "CU",
    dialCode: "+53",
  ),
  Country(
    name: "Κύπρος",
    code: "CY",
    dialCode: "+357",
  ),
  Country(
    name: "Česká republika",
    code: "CZ",
    dialCode: "+420",
  ),
  Country(
    name: "Danmark",
    code: "DK",
    dialCode: "+45",
  ),
  Country(
    name: "Djibouti",
    code: "DJ",
    dialCode: "+253",
  ),
  Country(
    name: "Dominica",
    code: "DM",
    dialCode: "+1767",
  ),
  Country(
    name: "República Dominicana",
    code: "DO",
    dialCode: "+1",
  ),
  Country(
    name: "Ecuador",
    code: "EC",
    dialCode: "+593",
  ),
  Country(
    name: "مصر‎",
    code: "EG",
    dialCode: "+20",
  ),
  Country(
    name: "El Salvador",
    code: "SV",
    dialCode: "+503",
  ),
  Country(
    name: "Guinea Ecuatorial",
    code: "GQ",
    dialCode: "+240",
  ),
  Country(
    name: "ኤርትራ",
    code: "ER",
    dialCode: "+291",
  ),
  Country(
    name: "Eesti",
    code: "EE",
    dialCode: "+372",
  ),
  Country(
    name: "ኢትዮጵያ",
    code: "ET",
    dialCode: "+251",
  ),
  Country(
    name: "Falkland Islands",
    code: "FK",
    dialCode: "+500",
  ),
  Country(
    name: "Føroyar",
    code: "FO",
    dialCode: "+298",
  ),
  Country(
    name: "Fiji",
    code: "FJ",
    dialCode: "+679",
  ),
  Country(
    name: "Suomi",
    code: "FI",
    dialCode: "+358",
  ),
  Country(
    name: "France",
    code: "FR",
    dialCode: "+33",
  ),
  Country(
    name: "Guyane française",
    code: "GF",
    dialCode: "+594",
  ),
  Country(
    name: "Polynésie française",
    code: "PF",
    dialCode: "+689",
  ),
  Country(
    name: "Territoire des Terres australes et antarctiques fr",
    code: "TF",
    dialCode: "+262",
  ),
  Country(
    name: "Gabon",
    code: "GA",
    dialCode: "+241",
  ),
  Country(
    name: "Gambia",
    code: "GM",
    dialCode: "+220",
  ),
  Country(
    name: "საქართველო",
    code: "GE",
    dialCode: "+995",
  ),
  Country(
    name: "Deutschland",
    code: "DE",
    dialCode: "+49",
  ),
  Country(
    name: "Ghana",
    code: "GH",
    dialCode: "+233",
  ),
  Country(
    name: "Gibraltar",
    code: "GI",
    dialCode: "+350",
  ),
  Country(
    name: "Ελλάδα",
    code: "GR",
    dialCode: "+30",
  ),
  Country(
    name: "Kalaallit Nunaat",
    code: "GL",
    dialCode: "+299",
  ),
  Country(
    name: "Grenada",
    code: "GD",
    dialCode: "+1473",
  ),
  Country(
    name: "Guadeloupe",
    code: "GP",
    dialCode: "+590",
  ),
  Country(
    name: "Guam",
    code: "GU",
    dialCode: "+1671",
  ),
  Country(
    name: "Guatemala",
    code: "GT",
    dialCode: "+502",
  ),
  Country(
    name: "Guernsey",
    code: "GG",
    dialCode: "+44",
  ),
  Country(
    name: "Guinée",
    code: "GN",
    dialCode: "+224",
  ),
  Country(
    name: "Guiné-Bissau",
    code: "GW",
    dialCode: "+245",
  ),
  Country(
    name: "Guyana",
    code: "GY",
    dialCode: "+592",
  ),
  Country(
    name: "Haïti",
    code: "HT",
    dialCode: "+509",
  ),
  Country(
    name: "Heard Island and McDonald Islands",
    code: "HM",
    dialCode: "+0",
  ),
  Country(
    name: "Vaticano",
    code: "VA",
    dialCode: "+379",
  ),
  Country(
    name: "Honduras",
    code: "HN",
    dialCode: "+504",
  ),
  Country(
    name: "香港",
    code: "HK",
    dialCode: "+852",
  ),
  Country(
    name: "Magyarország",
    code: "HU",
    dialCode: "+36",
  ),
  Country(
    name: "Ísland",
    code: "IS",
    dialCode: "+354",
  ),
  Country(
    name: "भारत",
    code: "IN",
    dialCode: "+91",
  ),
  Country(
    name: "Indonesia",
    code: "ID",
    dialCode: "+62",
  ),
  Country(
    name: "ایران",
    code: "IR",
    dialCode: "+98",
  ),
  Country(
    name: "العراق",
    code: "IQ",
    dialCode: "+964",
  ),
  Country(
    name: "Éire",
    code: "IE",
    dialCode: "+353",
  ),
  Country(
    name: "Isle of Man",
    code: "IM",
    dialCode: "+44",
  ),
  Country(
    name: "יִשְׂרָאֵל",
    code: "IL",
    dialCode: "+972",
  ),
  Country(
    name: "Italia",
    code: "IT",
    dialCode: "+39",
  ),
  Country(
    name: "Jamaica",
    code: "JM",
    dialCode: "+1876",
  ),
  Country(
    name: "日本",
    code: "JP",
    dialCode: "+81",
  ),
  Country(
    name: "Jersey",
    code: "JE",
    dialCode: "+44",
  ),
  Country(
    name: "الأردن",
    code: "JO",
    dialCode: "+962",
  ),
  Country(
    name: "Қазақстан",
    code: "KZ",
    dialCode: "+7",
  ),
  Country(
    name: "Kenya",
    code: "KE",
    dialCode: "+254",
  ),
  Country(
    name: "Kiribati",
    code: "KI",
    dialCode: "+686",
  ),
  Country(
    name: "북한",
    code: "KP",
    dialCode: "+850",
  ),
  Country(
    name: "대한민국",
    code: "KR",
    dialCode: "+82",
  ),
  Country(
    name: "Republika e Kosovës",
    code: "XK",
    dialCode: "+383",
  ),
  Country(
    name: "الكويت",
    code: "KW",
    dialCode: "+965",
  ),
  Country(
    name: "Кыргызстан",
    code: "KG",
    dialCode: "+996",
  ),
  Country(
    name: "ສປປລາວ",
    code: "LA",
    dialCode: "+856",
  ),
  Country(
    name: "Latvija",
    code: "LV",
    dialCode: "+371",
  ),
  Country(
    name: "لبنان",
    code: "LB",
    dialCode: "+961",
  ),
  Country(
    name: "Lesotho",
    code: "LS",
    dialCode: "+266",
  ),
  Country(
    name: "Liberia",
    code: "LR",
    dialCode: "+231",
  ),
  Country(
    name: "‏ليبيا",
    code: "LY",
    dialCode: "+218",
  ),
  Country(
    name: "Liechtenstein",
    code: "LI",
    dialCode: "+423",
  ),
  Country(
    name: "Lietuva",
    code: "LT",
    dialCode: "+370",
  ),
  Country(
    name: "Luxembourg",
    code: "LU",
    dialCode: "+352",
  ),
  Country(
    name: "澳門",
    code: "MO",
    dialCode: "+853",
  ),
  Country(
    name: "Македонија",
    code: "MK",
    dialCode: "+389",
  ),
  Country(
    name: "Madagasikara",
    code: "MG",
    dialCode: "+261",
  ),
  Country(
    name: "Malawi",
    code: "MW",
    dialCode: "+265",
  ),
  Country(
    name: "Malaysia",
    code: "MY",
    dialCode: "+60",
  ),
  Country(
    name: "Maldives",
    code: "MV",
    dialCode: "+960",
  ),
  Country(
    name: "Mali",
    code: "ML",
    dialCode: "+223",
  ),
  Country(
    name: "Malta",
    code: "MT",
    dialCode: "+356",
  ),
  Country(
    name: "M̧ajeļ",
    code: "MH",
    dialCode: "+692",
  ),
  Country(
    name: "Martinique",
    code: "MQ",
    dialCode: "+596",
  ),
  Country(
    name: "موريتانيا",
    code: "MR",
    dialCode: "+222",
  ),
  Country(
    name: "Maurice",
    code: "MU",
    dialCode: "+230",
  ),
  Country(
    name: "Mayotte",
    code: "YT",
    dialCode: "+262",
  ),
  Country(
    name: "México",
    code: "MX",
    dialCode: "+52",
  ),
  Country(
    name: "Micronesia",
    code: "FM",
    dialCode: "+691",
  ),
  Country(
    name: "Moldova",
    code: "MD",
    dialCode: "+373",
  ),
  Country(
    name: "Monaco",
    code: "MC",
    dialCode: "+377",
  ),
  Country(
    name: "Монгол улс",
    code: "MN",
    dialCode: "+976",
  ),
  Country(
    name: "Црна Гора",
    code: "ME",
    dialCode: "+382",
  ),
  Country(
    name: "Montserrat",
    code: "MS",
    dialCode: "+1664",
  ),
  Country(
    name: "المغرب",
    code: "MA",
    dialCode: "+212",
  ),
  Country(
    name: "Moçambique",
    code: "MZ",
    dialCode: "+258",
  ),
  Country(
    name: "Myanma",
    code: "MM",
    dialCode: "+95",
  ),
  Country(
    name: "Namibia",
    code: "NA",
    dialCode: "+264",
  ),
  Country(
    name: "Nauru",
    code: "NR",
    dialCode: "+674",
  ),
  Country(
    name: "नपल",
    code: "NP",
    dialCode: "+977",
  ),
  Country(
    name: "Nederland",
    code: "NL",
    dialCode: "+31",
  ),
  Country(
    name: "Netherlands Antilles",
    code: "AN",
    dialCode: "+599",
  ),
  Country(
    name: "Nouvelle-Calédonie",
    code: "NC",
    dialCode: "+687",
  ),
  Country(
    name: "New Zealand",
    code: "NZ",
    dialCode: "+64",
  ),
  Country(
    name: "Nicaragua",
    code: "NI",
    dialCode: "+505",
  ),
  Country(
    name: "Niger",
    code: "NE",
    dialCode: "+227",
  ),
  Country(
    name: "Nigeria",
    code: "NG",
    dialCode: "+234",
  ),
  Country(
    name: "Niuē",
    code: "NU",
    dialCode: "+683",
  ),
  Country(
    name: "Norfolk Island",
    code: "NF",
    dialCode: "+672",
  ),
  Country(
    name: "Northern Mariana Islands",
    code: "MP",
    dialCode: "+1670",
  ),
  Country(
    name: "Norge",
    code: "NO",
    dialCode: "+47",
  ),
  Country(
    name: "عمان",
    code: "OM",
    dialCode: "+968",
  ),
  Country(
    name: "Pakistan",
    code: "PK",
    dialCode: "+92",
  ),
  Country(
    name: "Palau",
    code: "PW",
    dialCode: "+680",
  ),
  Country(
    name: "فلسطين",
    code: "PS",
    dialCode: "+970",
  ),
  Country(
    name: "Panamá",
    code: "PA",
    dialCode: "+507",
  ),
  Country(
    name: "Papua Niugini",
    code: "PG",
    dialCode: "+675",
  ),
  Country(
    name: "Paraguay",
    code: "PY",
    dialCode: "+595",
  ),
  Country(
    name: "Perú",
    code: "PE",
    dialCode: "+51",
  ),
  Country(
    name: "Pilipinas",
    code: "PH",
    dialCode: "+63",
  ),
  Country(
    name: "Pitcairn Islands",
    code: "PN",
    dialCode: "+64",
  ),
  Country(
    name: "Polska",
    code: "PL",
    dialCode: "+48",
  ),
  Country(
    name: "Portugal",
    code: "PT",
    dialCode: "+351",
  ),
  Country(
    name: "Puerto Rico",
    code: "PR",
    dialCode: "+1939",
  ),
  Country(
    name: "Puerto Rico",
    code: "PR",
    dialCode: "+1787",
  ),
  Country(
    name: "قطر",
    code: "QA",
    dialCode: "+974",
  ),
  Country(
    name: "România",
    code: "RO",
    dialCode: "+40",
  ),
  Country(
    name: "Россия",
    code: "RU",
    dialCode: "+7",
  ),
  Country(
    name: "Rwanda",
    code: "RW",
    dialCode: "+250",
  ),
  Country(
    name: "La Réunion",
    code: "RE",
    dialCode: "+262",
  ),
  Country(
    name: "Saint-Barthélemy",
    code: "BL",
    dialCode: "+590",
  ),
  Country(
    name: "Saint Helena",
    code: "SH",
    dialCode: "+290",
  ),
  Country(
    name: "Saint Kitts and Nevis",
    code: "KN",
    dialCode: "+1869",
  ),
  Country(
    name: "Saint Lucia",
    code: "LC",
    dialCode: "+1758",
  ),
  Country(
    name: "Saint-Martin",
    code: "MF",
    dialCode: "+590",
  ),
  Country(
    name: "Saint-Pierre-et-Miquelon",
    code: "PM",
    dialCode: "+508",
  ),
  Country(
    name: "Saint Vincent and the Grenadines",
    code: "VC",
    dialCode: "+1784",
  ),
  Country(
    name: "Samoa",
    code: "WS",
    dialCode: "+685",
  ),
  Country(
    name: "San Marino",
    code: "SM",
    dialCode: "+378",
  ),
  Country(
    name: "São Tomé e Príncipe",
    code: "ST",
    dialCode: "+239",
  ),
  Country(
    name: "العربية السعودية",
    code: "SA",
    dialCode: "+966",
  ),
  Country(
    name: "Sénégal",
    code: "SN",
    dialCode: "+221",
  ),
  Country(
    name: "Србија",
    code: "RS",
    dialCode: "+381",
  ),
  Country(
    name: "Seychelles",
    code: "SC",
    dialCode: "+248",
  ),
  Country(
    name: "Sierra Leone",
    code: "SL",
    dialCode: "+232",
  ),
  Country(
    name: "Singapore",
    code: "SG",
    dialCode: "+65",
  ),
  Country(
    name: "Slovensko",
    code: "SK",
    dialCode: "+421",
  ),
  Country(
    name: "Slovenija",
    code: "SI",
    dialCode: "+386",
  ),
  Country(
    name: "Solomon Islands",
    code: "SB",
    dialCode: "+677",
  ),
  Country(
    name: "Soomaaliya",
    code: "SO",
    dialCode: "+252",
  ),
  Country(
    name: "South Africa",
    code: "ZA",
    dialCode: "+27",
  ),
  Country(
    name: "South Sudan",
    code: "SS",
    dialCode: "+211",
  ),
  Country(
    name: "South Georgia",
    code: "GS",
    dialCode: "+500",
  ),
  Country(
    name: "España",
    code: "ES",
    dialCode: "+34",
  ),
  Country(
    name: "śrī laṃkāva",
    code: "LK",
    dialCode: "+94",
  ),
  Country(
    name: "السودان",
    code: "SD",
    dialCode: "+249",
  ),
  Country(
    name: "Suriname",
    code: "SR",
    dialCode: "+597",
  ),
  Country(
    name: "Svalbard og Jan Mayen",
    code: "SJ",
    dialCode: "+47",
  ),
  Country(
    name: "Swaziland",
    code: "SZ",
    dialCode: "+268",
  ),
  Country(
    name: "Sverige",
    code: "SE",
    dialCode: "+46",
  ),
  Country(
    name: "Schweiz",
    code: "CH",
    dialCode: "+41",
  ),
  Country(
    name: "سوريا",
    code: "SY",
    dialCode: "+963",
  ),
  Country(
    name: "臺灣",
    code: "TW",
    dialCode: "+886",
  ),
  Country(
    name: "Тоҷикистон",
    code: "TJ",
    dialCode: "+992",
  ),
  Country(
    name: "Tanzania",
    code: "TZ",
    dialCode: "+255",
  ),
  Country(
    name: "ประเทศไทย",
    code: "TH",
    dialCode: "+66",
  ),
  Country(
    name: "Timor-Leste",
    code: "TL",
    dialCode: "+670",
  ),
  Country(
    name: "Togo",
    code: "TG",
    dialCode: "+228",
  ),
  Country(
    name: "Tokelau",
    code: "TK",
    dialCode: "+690",
  ),
  Country(
    name: "Tonga",
    code: "TO",
    dialCode: "+676",
  ),
  Country(
    name: "Trinidad and Tobago",
    code: "TT",
    dialCode: "+1868",
  ),
  Country(
    name: "تونس",
    code: "TN",
    dialCode: "+216",
  ),
  Country(
    name: "Türkiye",
    code: "TR",
    dialCode: "+90",
  ),
  Country(
    name: "Türkmenistan",
    code: "TM",
    dialCode: "+993",
  ),
  Country(
    name: "Turks and Caicos Islands",
    code: "TC",
    dialCode: "+1649",
  ),
  Country(
    name: "Tuvalu",
    code: "TV",
    dialCode: "+688",
  ),
  Country(
    name: "Uganda",
    code: "UG",
    dialCode: "+256",
  ),
  Country(
    name: "Україна",
    code: "UA",
    dialCode: "+380",
  ),
  Country(
    name: "دولة الإمارات العربية المتحدة",
    code: "AE",
    dialCode: "+971",
  ),
  Country(
    name: "United Kingdom",
    code: "GB",
    dialCode: "+44",
  ),
  Country(
    name: "United States",
    code: "US",
    dialCode: "+1",
  ),
  Country(
    name: "Uruguay",
    code: "UY",
    dialCode: "+598",
  ),
  Country(
    name: "O‘zbekiston",
    code: "UZ",
    dialCode: "+998",
  ),
  Country(
    name: "Vanuatu",
    code: "VU",
    dialCode: "+678",
  ),
  Country(
    name: "Venezuela",
    code: "VE",
    dialCode: "+58",
  ),
  Country(
    name: "Việt Nam",
    code: "VN",
    dialCode: "+84",
  ),
  Country(
    name: "British Virgin Islands",
    code: "VG",
    dialCode: "+1284",
  ),
  Country(
    name: "United States Virgin Islands",
    code: "VI",
    dialCode: "+1340",
  ),
  Country(
    name: "Wallis et Futuna",
    code: "WF",
    dialCode: "+681",
  ),
  Country(
    name: "اليَمَن",
    code: "YE",
    dialCode: "+967",
  ),
  Country(
    name: "Zambia",
    code: "ZM",
    dialCode: "+260",
  ),
  Country(
    name: "Zimbabwe",
    code: "ZW",
    dialCode: "+263",
  ),
];
